service/rpc/share.go

```diff
@@ -0,0 +1,77 @@
+package rpc
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/celestiaorg/celestia-node/service/header"
+)
+
+const namespacedSharesEndpoint = "/namespaced_shares"
```

Comment:
Where and how will these endpoints be documented?

Reply:
I've provided docs of these endpoints in the description of the PR which will then be transcribed to API docs on official website

cc @YazzyYaz

Reply:
I think it would be nice if the docs were in the code and you could run a script to generate markdown docs (for the web-site and godocs) from that. Kinda orthogonal for this PR but the more indirections and the further away the docs from the code (currently in the opening comment, later on a web site), the less likely the docs will stay up to date. Unless we check against the examples in the docs automatically.

Reply:
We should have a suite that tests exposed endpoints and checks the response. This is for later though. For now, manually testing endpoints is good as we have active users (Tomasz).

Reply:
https://github.com/celestiaorg/celestia-node/pull/693/commits/ec640db3656d9a1c7bec0d0d495390ed3badf493